### DISCUSSION TITLE
Confusion about history code ev_io_start when read EAGAIN

### DISCUSSION THREAD

**Discussion by ylme** [libuv/libuv#3330] [Q&A] [answered]
Hi all
  See the [commit](https://github.com/libuv/libuv/commit/42cefd934e4275e3614d4900f2f5c0d35cdfbdf2#diff-d9bf5cba3ffed4a7cebae60c0316c583497f4879e251ba6c7f6f876deed7da69R334) line 334, I think the call to `ev_io_start` should make no sense. I'm confused abort what is the purpose here. I hope to find some directions, thanks.

**Comment by bnoordhuis** [comment]
I'm kind of curious why you link to a 10 year old commit. Libuv hasn't had a dependency on libev for almost as long as that.

**Reply by ylme** [reply]
Yes, libuv hasn't depended on libev for so many years. I just want to learn how to use libev by read the code the libuv how to use in the history version. Then I read the consufed call to `ev_io_start` when `read` occur `EAGAIN`. It's interesting here, because we usually ignore `EAGAIN` or `EWOULDBLOCK` on a non-blocking fd. And libuv still does the [similar thing](https://github.com/libuv/libuv/blob/v1.x/src/unix/stream.c#L1189) , It's worth learning, so I want to figure it out.

**Comment by bnoordhuis** [comment, marked as answer]
Oh, that `uv__io_start()` is _probably_ superfluous but harmless. It's a no-op when the file descriptor is already being polled for readability.

**Reply by ylme** [reply]
Got it, Thanks.
